### ISSUE TITLE
Be able to change their Minilock key / Password

### ISSUE THREAD

**Issue by IpsmLorem** [PeerioTechnologies/peerio-client#89]
It could be interresting to be able to change their Minilock ID / Password in the account tab.

People could change it on a regular basis.

**Comment by kaepora** [comment]
Yes, this is planned. I'm changing the title of this issue to something more relevant to Peerio.

**Comment by hdf** [comment]
But then all your encrypted files become unavailable. So maybe have a backup of all previous identities, and the system should remember which file belongs to witch entity? Also, you should be able to delete entities from the backup.

**Comment by IpsmLorem** [comment]
Identities is bound to your login name. The system can re encrypt the new xsalsa20 symetric key with the new Minilock ID and remove the old one.

**Comment by hdf** [comment]
So stuff isn't stored encrypted with your public key, only a symmetric key, and the symmetric key is stored encrypted with your public key?

**Comment by IpsmLorem** [comment]
I assumed that. 
I didn't deeply checked the minilock description to verify my assumption. I may be wrong.

**Comment by IpsmLorem** [comment]
Minillock file format use a unique fileKey encrypted for a permet secret key (and nonces)

**Comment by hdf** [comment]
So files would have to be re-encrypted if the pass phrase is changed? That's not good.

**Comment by kaepora** [comment]
Your direction is off-track. We already have a solution for this. Be patient!
